### PR TITLE
jclklib: Refactor servo lock status assignment in event_handle

### PR DESCRIPTION
**Pull request by JunAnnLaiIntel**
The previous implementation extracted the `servo_state` and used it to determine the `servo_locked` status. This has been refactored to directly use the `portState` to set `servo_locked`.

### CODE REVIEW

File: jclklib/proxy/connect_ptp4l.cpp [hunk 1]
```diff
@@ -120,14 +118,16 @@ void event_handle()
...
+            } else {
+                pe.servo_locked = portState >= SLAVE ? true:false;
             }
-
             break;
         case PORT_PROPERTIES_NP:
             /* Retrieve current port state when proxy is started */
             pd = (PORT_DATA_SET_t *)data;
             portState = pd->portState;
+            pe.servo_locked = portState >= SLAVE ? true:false;
```
Comment (yoongsiang2) [inline jclklib/proxy/connect_ptp4l.cpp:130]: use uncalibrated

File: jclklib/proxy/connect_ptp4l.cpp [hunk 2]
```diff
@@ -120,14 +118,16 @@ void event_handle()
             if (portState <= PASSIVE) {
                 pe.master_offset = 0;
                 memset(pe.gm_identity, 0, sizeof(pe.gm_identity));
-                pe.servo_locked = 0;
+                pe.servo_locked = false;
+            } else {
+                pe.servo_locked = portState >= SLAVE ? true:false;
```
Comment (yoongsiang2) [inline jclklib/proxy/connect_ptp4l.cpp:123]: pe.servo_locked =true;

**Review by yoongsiang2 — CHANGES_REQUESTED** [review]

File: uctest/json2msg.c [hunk 3]
```diff
@@ -1982,7 +1982,6 @@ Test(Json2msgTest, TIME_STATUS_NP)
             "\"fractional_nanoseconds\":0,"
             "\"gmPresent\":0,"
             "\"gmIdentity\":\"c47d46.fffe.20acae\","
```
Comment (yoongsiang2) [inline uctest/json2msg.c:1984]: need to remove , 
"\"gmIdentity\":\"c47d46.fffe.20acae\""

File: uctest/msg2json.c [hunk 4]
```diff
@@ -1638,7 +1637,6 @@ Test(Tlv2JsonTest, TIME_STATUS_NP)
             "  \"fractional_nanoseconds\" : 0,\n"
             "  \"gmPresent\" : 0,\n"
             "  \"gmIdentity\" : \"c47d46.fffe.20acae\",\n"
```
Comment (yoongsiang2) [inline uctest/msg2json.c:1639]: need to remove ,
"  \"gmIdentity\" : \"c47d46.fffe.20acae\"\n"

File: utest/json2msg.cpp [hunk 5]
```diff
@@ -1841,7 +1841,6 @@ TEST(Json2msgTest, TIME_STATUS_NP)
             "\"fractional_nanoseconds\":0,"
             "\"gmPresent\":0,"
             "\"gmIdentity\":\"c47d46.fffe.20acae\","
```
Comment (yoongsiang2) [inline utest/json2msg.cpp:1843]: remove ,

File: utest/msg2json.cpp [hunk 6]
```diff
@@ -1332,7 +1331,6 @@ TEST(Tlv2JsonTest, TIME_STATUS_NP)
         "  \"fractional_nanoseconds\" : 0,\n"
         "  \"gmPresent\" : 0,\n"
         "  \"gmIdentity\" : \"c47d46.fffe.20acae\",\n"
```
Comment (yoongsiang2) [inline utest/msg2json.cpp:1333]: remove ,

File: utest/pmc_dump.cpp [hunk 7]
```diff
@@ -776,7 +775,6 @@ TEST(PmcDumpTest, TIME_STATUS_NP)
         IDENT "lastGmPhaseChange          0x0000'0000000000000000.0000"
         IDENT "gmPresent                  false"
         IDENT "gmIdentity                 c47d46.fffe.20acae"
```
Comment (yoongsiang2) [inline utest/pmc_dump.cpp:777]: add );

File: jclklib/proxy/connect_ptp4l.cpp [hunk 8]
```diff
@@ -120,14 +118,16 @@ void event_handle()
             if (portState <= PASSIVE) {
```
Comment (yoongsiang2) [inline jclklib/proxy/connect_ptp4l.cpp:118]: use time receiver

File: jclklib/proxy/connect_ptp4l.cpp [hunk 9]
```diff
@@ -120,14 +118,16 @@ void event_handle()
...
+            } else {
+                pe.synced_to_primary_clock = true;
             }
-
             break;
         case PORT_PROPERTIES_NP:
             /* Retrieve current port state when proxy is started */
             pd = (PORT_DATA_SET_t *)data;
             portState = pd->portState;
+            pe.synced_to_primary_clock = portState >= SLAVE ? true:false;
```
Comment (yoongsiang2) [inline jclklib/proxy/connect_ptp4l.cpp:130]: use time receiver

**Review by yoongsiang2 — CHANGES_REQUESTED** [review]

**Comment by yoongsiang2** [comment]
Add new patch in your PR to resolve license issue, so that we can check whether CI is pass or fail

diff --git a/.reuse/dep5 b/.reuse/dep5
index 0b2e4e3..81aa787 100644
--- a/.reuse/dep5
+++ b/.reuse/dep5
@@ -9,6 +9,10 @@ Files: */changelog
 Copyright: 2021 © Erez Geva
 License: GFDL-1.3-no-invariants-or-later

+Files: jclklib/image/*.png
+Copyright: 2024 © Intel Corporation
+License: GFDL-1.3-no-invariants-or-later
+
 Files: debian/co* debian/source/format
 Copyright: 2021 © Erez Geva
 License: GPL-3.0-or-later

File: jclklib/proxy/connect_ptp4l.cpp (hunk 9 above)
Comment (AzuraTarmiziIntel) [reply jclklib/proxy/connect_ptp4l.cpp]: I got this from fsm.h ( linux-ptp)

/** Defines the state of a port. */
enum port_state {
	PS_INITIALIZING = 1,
	PS_FAULTY,
	PS_DISABLED,
	PS_LISTENING,
	PS_PRE_MASTER,
	PS_MASTER,
	PS_PASSIVE,
	PS_UNCALIBRATED,
	PS_SLAVE,
	PS_GRAND_MASTER, /*non-standard extension*/ 
}; 

Maybe we shd stick with SLAVE ?  to be aligned with linux-ptp naming. I understand the concern with name we shouldnt use. 


Future note : 
I wonder why libptpmgmt didnt use directly the linux-ptp header  (fsm.h)

File: jclklib/proxy/connect_ptp4l.cpp [hunk 10]
```diff
@@ -115,7 +115,7 @@ void event_handle()
             portState = pd->portState;
 
             /* Reset TIME_STATUS_NP data if port_state <= PASSIVE */
-            if (portState <= PASSIVE) {
+            if (portState <= TIME_RECEIVER) {
```
Comment (AzuraTarmiziIntel) [inline jclklib/proxy/connect_ptp4l.cpp:118]: need to align. slave or time_receiver . semua mesti sama.

File: jclklib/client/notification_msg.cpp [hunk 11]
```diff
@@ -62,8 +62,10 @@ void ClientNotificationMessage::addClientState(ClientState *newClientState)
 
 void ClientNotificationMessage::deleteClientState(ClientState *newClientState)
 {
-    ClientStateArray.erase(std::remove(ClientStateArray.begin(),
-        ClientStateArray.end(), newClientState), ClientStateArray.end());
+    auto it = std::remove(ClientStateArray.begin(), ClientStateArray.end(),
+                          newClientState);
```
Comment (AzuraTarmiziIntel) [inline jclklib/client/notification_msg.cpp:66]: Just curious why the change is needed ?

**Comment by AzuraTarmiziIntel** [comment]
Hi peter, 

great work. 

I would like to request for the first change to be squashed with the second commit (servo_state changes) 

Because the idea is to have at each commit - the compilation should pass. this is to avoid difficult cherry-picking in later stage.

File: jclklib/client/notification_msg.cpp (hunk 11 above)
Comment (JunAnnLaiIntel) [reply jclklib/client/notification_msg.cpp:66]: This change is to fix the error that found in CI, as shown in below. But this change does not solve the issue, I am working on another solution.

notification_msg.cpp: In static member function ‘static void JClkLibClient::ClientNotificationMessage::deleteClientState(JClkLibClient::ClientState*)’:
notification_msg.cpp:65:49: error: cannot convert ‘std::vector<JClkLibClient::ClientState*>::iterator’ to ‘const char*’
   65 |     auto it = std::remove(ClientStateArray.begin(), ClientStateArray.end(),
      |                           ~~~~~~~~~~~~~~~~~~~~~~^~
      |                                                 |
      |                                                 std::vector<JClkLibClient::ClientState*>::iterator
In file included from /usr/include/c++/14.1.1/cstdio:42,
                 from /usr/include/c++/14.1.1/ext/string_conversions.h:45,
                 from /usr/include/c++/14.1.1/bits/basic_string.h:4154,
                 from /usr/include/c++/14.1.1/string:54,
                 from /__w/libptpmgmt_iaclocklib/libptpmgmt_iaclocklib/jclklib/client/../client/jclk_client_state.hpp:19,
                 from notification_msg.cpp:15:
/usr/include/stdio.h:158:32: note:   initializing argument 1 of ‘int remove(const char*)’
  158 | extern int remove (const char *__filename) __THROW;
      |                    ~~~~~~~~~~~~^~~~~~~~~~

File: jclklib/proxy/connect_ptp4l.cpp (hunk 9 above)
Comment (JunAnnLaiIntel) [reply jclklib/proxy/connect_ptp4l.cpp]: Just for reference:
/** Port state */
enm(portState_e) sz(: uint8_t) {
    NM(INITIALIZING)         = 1, /**< Initializing */
    NM(FAULTY)               = 2, /**< Faulty */
    NM(DISABLED)             = 3, /**< Disabled */
    NM(LISTENING)            = 4, /**< Listening */
    NM(PRE_MASTER)           = 5, /**< Pre timeTransmitter */
    NM(PRE_TIME_TRANSMITTER) = 5, /**< Pre timeTransmitter */
    NM(MASTER)               = 6, /**< TimeTransmitter */
    NM(TIME_TRANSMITTER)     = 6, /**< TimeTransmitter */
    NM(PASSIVE)              = 7, /**< Passive */
    NM(UNCALIBRATED)         = 8, /**< Uncalibrated */
    NM(SLAVE)                = 9, /**< TimeReceiver */
    NM(TIME_RECEIVER)        = 9, /**< TimeReceiver */
};

File: jclklib/proxy/connect_ptp4l.cpp (hunk 10 above)
Comment (JunAnnLaiIntel) [reply jclklib/proxy/connect_ptp4l.cpp:118]: Agree.

**Review by yoongsiang2 — APPROVED** [review]
LGTM

File: jclklib/proxy/connect_ptp4l.cpp [hunk 12]
```diff
@@ -117,17 +115,19 @@ void event_handle()
             portState = pd->portState;
 
             /* Reset TIME_STATUS_NP data if port_state <= PASSIVE */
-            if (portState <= PASSIVE) {
+            if (portState <= SLAVE) {
```
Comment (yoongsiang2) [inline jclklib/proxy/connect_ptp4l.cpp:118]: should be portState < SLAVE

**Review by yoongsiang2 — CHANGES_REQUESTED** [review]

**Review by yoongsiang2 — APPROVED** [review]
LGTM
